screen.c

```diff
@@ -60,16 +71,30 @@ screen_reinit(struct screen *s)
 	grid_clear_lines(s->grid, s->grid->hsize, s->grid->sy, 8);
 
 	screen_clear_selection(s);
+
+	while ((title_entry = TAILQ_FIRST(s->titles)) != NULL) {
+		TAILQ_REMOVE(s->titles, title_entry, entry);
```

Comment:
I know this is based on the original code, but can you check if `TAILQ_FOREACH_SAFE` and hence `TAILQ_REMOVE` does the same thing?  If so, I'd suggest switching over to that.

Reply:
`TAILQ_FOREACH_SAFE` and `TAILQ_REMOVE` should work fine.